src/modules/iotjs_module_buffer.c

```diff
@@ -252,6 +252,17 @@ JHANDLER_FUNCTION(Compare) {
 }
 
 
+#define CONVERT_DOUBLE_TO_SIZE(n, d) \
+  size_t n;                          \
+  do {                               \
+    if (d < 0) {                     \
```

Comment:
It's undefined, so we should handle it. http://stackoverflow.com/questions/3986795/what-is-the-result-of-casting-float-inf-inf-and-nan-to-integer-in-c
Any suggestion? What about using `0`, `INT64_MIN` and `INT64_MAX` for NaN, -inf and inf respectively?